### ISSUE TITLE
MSEloss with reduce=False returns a tensor of same size as the input and target  (rather than just one loss per batch element)

### ISSUE THREAD

**Issue by dineshj1** [pytorch/pytorch#4621]
MSELoss with reduce=False doesn't behave as documented.
```
loss = nn.MSELoss(reduce=False)
input = autograd.Variable(torch.randn(3, 5)) # 3 elements in batch of size 5
target = autograd.Variable(torch.randn(3, 5))
output = loss(input, target)
print(output)
```
As documented, this should return one loss for each of the 3 batch elements. Instead, it returns:
```
Variable containing:
  5.4552   2.3369   3.3013   0.0396  13.2650
  1.4288   0.4156   0.3027   0.3006   0.2276
  7.6768   1.0778   0.1317   1.9284   3.4640
[torch.FloatTensor of size 3x5]
```

A similar setup with with CrossEntropyLoss(reduce=False) instead of MSELoss correctly returns a FloatTensor of size 3.
